More guidance on idle timeouts (#102)
quicwg/ops-drafts: draft-ietf-quic-applicability.md

STOP_SENDING frames provide similar capablities. Usually application error codes
are defined to be applicabile to all three frames.

Alternatively, a QUIC connection can be silently closed by each endpoint

> separately after an idle timeout. If enabled as indcated by a transport

Reviewer: ```suggestion
separately after an idle timeout. If enabled as indicated by a transport
```

parameter in the handshake, the idle timeout is announced for each
endpoint during connection establishment and the effective value for this

> connection is the minimum of these two advertised value. An application

Reviewer: ```suggestion
connection is the minimum of the two advertised values. An application
```

therefore should be able to configure its own maximum value as well as

> have access to the computed minimum value for this connection. An application

Reviewer 2: I don’t understand why an implementation needs access to the computed value. A reasonable QUIC implementation could take the application’s idle timeout and then take care of sending PINGs to the peer to keep the connection open.

Reviewer 3: The text here, I think, is appropriate for general-purpose QUIC implementations (atop which multiple applications could be bound), but yeah, it could be either-or for applications tightly bound to their implementation, based on application keepalive logic.

Reviewer: I think that - at the level of this document - this is fine advice.  Applications do benefit from knowing when a timeout is imminent because they might want to manage the risk of failure for transactions that can't be easily retried.  At or near the timeout is not a great time to start something.

may adjust the maximum idle timeout based on the number of open or expected
connections as shorter timeout values may free-up memory more quickly.
If an application desires to keep the connection open for longer
than the announced timeout, it can send keep-alives messages, or a QUIC

> implementation may provide an option to defer the time-out to avoid unnesseary

Reviewer 2: Not every application protocol has a keep-alive option. I view keep-alives as the responsibility of the QUIC stack instead.

Author: Both might have benefits depending on the application, e.g. if you anyway need to exchange certain status update messages on application layer. Should be more explicit about that in the text?

> implementation may provide an option to defer the time-out to avoid unnesseary

Reviewer: ```suggestion
implementation may provide an option to defer the time-out to avoid unnecessary
```

> load, as specified in Section 10.2.2 of {{QUIC}}.

Reviewer: Be aware that this anchor is likely to change to 10.1.2.

> See {#resumption-v-keepalive} for further guidance on keep-alives.

Reviewer: ```suggestion
See {{resumption-v-keepalive}} for further guidance on keep-alives.
```
